### PR TITLE
Fix #4774: Properly track version changes in JS methods

### LINKED ISSUE #4774: Emitter assumes exports can only change if the class changes
#4786 which we just merged breaks this assumption.

How to break it:

HelloWorld.scala

```
object A {
  @JSExport
  def testexport = B.test
}

object HelloWorld {
  def main(args: Array[String]): Unit = {
    println(A.asInstanceOf[js.Dynamic].testexport)
  }
}
```

B.scala

```diff
  object B {
-   def test = 2
+   def test = bar

    @noinline
    def bar = 3
  }
```

1. Compile & link
2. Outputs "2"
3. Apply the diff to B.scala
4. Still outputs "2"
5. Clean & build
6. Outputs "3"

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

File: linker/shared/src/test/scala/org/scalajs/linker/IncrementalTest.scala [hunk 1]
```diff
@@ -339,6 +339,39 @@ class IncrementalTest {
...
+      ),
+      v0 -> classDef(
+          AModule,
+          kind = ClassKind.ModuleClass,
+          superClass = Some(ObjectClass),
+          memberDefs = List(
+              trivialCtor(AModule),
+              JSMethodDef(EMF, str("foo"), Nil, None,
+                  Apply(EAF, LoadModule(BModule), targetMethodName, Nil)(IntType))(EOH, UNV)
+          )),
```
Comment (sjrd) [inline linker/shared/src/test/scala/org/scalajs/linker/IncrementalTest.scala:362]:
```suggestion
          )
      ),
```

File: linker/shared/src/test/scala/org/scalajs/linker/IncrementalTest.scala [hunk 2]
```diff
@@ -339,6 +339,39 @@ class IncrementalTest {
...
+      v(pre) -> classDef(
+          BModule,
+          kind = ClassKind.ModuleClass,
+          superClass = Some(ObjectClass),
+          memberDefs = List(
+              trivialCtor(BModule),
+              MethodDef(EMF, targetMethodName, NON, Nil, IntType,
+                  Some(int(if (pre) 1 else 2)))(EOH.withInline(true), UNV)))
```
Comment (sjrd) [inline linker/shared/src/test/scala/org/scalajs/linker/IncrementalTest.scala:370]:
```suggestion
                  Some(int(if (pre) 1 else 2)))(EOH.withInline(true), UNV)
          )
      )
```

File: linker/shared/src/test/scala/org/scalajs/linker/IncrementalTest.scala [hunk 3]
```diff
@@ -372,6 +372,47 @@ class IncrementalTest {
...
+      ),
+      v0 -> classDef(
+          AClass,
+          kind = ClassKind.JSClass,
+          superClass = Some(JSObject),
+          memberDefs = List(
+              JSConstructorDef(EMF.withNamespace(MemberNamespace.Constructor), Nil, None,
+                  JSConstructorBody(Nil, JSSuperConstructorCall(Nil), List({
+                    consoleLog(Apply(EAF, LoadModule(BModule), targetMethodName, Nil)(IntType))
+                  })))(EOH, UNV))),
```
Comment (sjrd) [inline linker/shared/src/test/scala/org/scalajs/linker/IncrementalTest.scala:403]: Same here.

File: linker/shared/src/test/scala/org/scalajs/linker/IncrementalTest.scala [hunk 4]
```diff
@@ -372,6 +372,47 @@ class IncrementalTest {
...
+      v(pre) -> classDef(
+          BModule,
+          kind = ClassKind.ModuleClass,
+          superClass = Some(ObjectClass),
+          memberDefs = List(
+              trivialCtor(BModule),
+              MethodDef(EMF, targetMethodName, NON, Nil, IntType,
+                  Some(int(if (pre) 1 else 2)))(EOH.withInline(true), UNV)))
```
Comment (sjrd) [inline linker/shared/src/test/scala/org/scalajs/linker/IncrementalTest.scala:411]: Same here.

**Review by sjrd — CHANGES_REQUESTED** [review]
Nits only. There's also a Scalastyle issue.

**Comment by gzm0** [comment]
Updated.

**Review by sjrd — APPROVED** [review]

**Pull request merged**